contracts/shared/Swap/OneInch/OneInchUniswapV3.sol

```diff
@@ -47,11 +47,9 @@ contract OneInchUniswapV3 is ISwapper {
   ) public payable override returns (uint256 amountOut) {
     // transfer the funds to be swapped from the sender into this contract
     TransferHelper.safeTransferFrom(_fromAsset, msg.sender, address(this), _amountIn);
-
     // decode the swap data
-    // the data included with the swap encodes with the selector so we need to remove it
     // https://docs.1inch.io/docs/aggregation-protocol/smart-contract/UnoswapV3Router#uniswapv3swap
-    (, uint256 _minReturn, uint256[] memory _pools) = abi.decode(_swapData[4:], (uint256, uint256, uint256[]));
```

Comment:
Did this need to change? I thought the 1inch swap data includes the selector.

Reply:
Yes right. 1inch swap data includes function selector which is `uniswapV3Swap(uint256,uint256,uint256[])`. This didn't make sense to me to have it included in swapData.

Reply:
But when it's generated from the API it includes the selector, that's why it's there. Otherwise it would need to be decoded and re-encoded again.